cwms-data-api/src/main/java/cwms/cda/data/dao/MeasurementDao.java

```diff
@@ -132,6 +138,28 @@ public void storeMeasurements(List<Measurement> measurements, boolean failIfExis
         });
     }
 
+    /**
+     * Updates an existing measurement
+     * @param measurement - the measurement to update
+     */
+    public void updateMeasurement(Measurement measurement) {
+        verifyMeasurementExists(measurement);
+        storeMeasurement(measurement, false);
+    }
+
+    /**
+     * Updates a list of existing measurements
+     * @param measurements - the measurements to update
+     */
+    public void updateMeasurements(List<Measurement> measurements)
+    {
+        for(Measurement measurement : measurements)
+        {
+            verifyMeasurementExists(measurement);
```

Comment:
ok, so I looked over this again and I understand a bit more.  I think you'll want to do a custom query checking against the location and the numbers available for that location.  so iterate over the unique locations, collect all of the numbers for each location, then you'd do something like `select * from AV_STREAMFLOW_MEAS where LOCATION_ID = '{your loc}' and OFFICE_ID = '{loc_office_id}' and NUMBER in [{your collection of numbers as strings}]` and then compare the lengths of the rows found to see if you found everything.  If not, then you collect the missing numbers and report them.  You could even report them per location.  That would give far more detail than a single exception for one measurement, and it would be 1 db call per unique location.

Reply:
Updated to use jooq query of specific number match to compare against